docs/specification/new-spec/consensus.md

```diff
@@ -0,0 +1,210 @@
+}
+```
+### Proposal
+Proposal contains height and round for which this proposal is made, BlockID as a unique identifier of proposed
+block, timestamp, and two fields (POLRound and POLBlockID) that are needed for termination of the consensus.
+The message is signed by the validator private key.  
+
+```
+type Proposal struct {
+	Height           int64            
+	Round            int              
+	Timestamp        time.Time        
+	BlockID          BlockID      
+	POLRound         int              
+	POLBlockID       BlockID          
+	Signature        crypto.Signature 
+}
+```
+
+NOTE: In the current version of the Tendermint, the consensus value in proposal is represented with
+PartSetHeader, and with BlockID in vote message. It should be aligned as suggested in this spec as 
+BlockID contains PartSetHeader.   
+
+## VoteMessage
+VoteMessage is sent to vote for some block (or to inform others that a process does not vote in the current round).
+
+```
+type VoteMessage struct {
+	Vote *types.Vote
+}
+```
+
+### Vote
+Vote contains validator's information (validator address and index), height and round for which the vote is sent,
+vote type, blockID if process vote for some block (`nil` otherwise) and a timestamp when the vote is sent. 
+The message is signed by the validator private key.
+
+```
+type Vote struct {
+	ValidatorAddress data.Bytes       
+	ValidatorIndex   int              
+	Height           int64            
+	Round            int              
+	Timestamp        time.Time        
+	Type             byte             
+	BlockID          BlockID          
+	Signature        crypto.Signature 
+}
+```
+## BlockPartMessage
+BlockPartMessage is sent when gossipping a piece of the proposed block. It contains height, round 
+and the block part. 
+
+```
+type BlockPartMessage struct {
+	Height int64
+	Round  int
+	Part   *types.Part
+}
+```
+
+## ProposalHeartbeatMessage 
+ProposalHeartbeatMessage is sent to signal that a node is alive and waiting for transactions 
+to be able to create a next block proposal.
+
+```
+type ProposalHeartbeatMessage struct {
+    Heartbeat *types.Heartbeat
+}
+```
+
+### Heartbeat
+Heartbeat contains validator information (address and index),
+height, round and sequence number (?). It is signed by the private key of the validator. 
+
+```
+type Heartbeat struct {
+	ValidatorAddress data.Bytes       
+	ValidatorIndex   int              
+	Height           int64            
+	Round            int              
+	Sequence         int              
+	Signature        crypto.Signature 
+}
+```
+
+## NewRoundStepMessage
+NewRoundStepMessage is sent for every step transition during the core consensus algorithm execution. It is 
+used in the gossip part of the Tendermint protocol to inform peers about a current height/round/step 
+a process is in.
+
+```
+type NewRoundStepMessage struct {
+	Height                int64
+	Round                 int
+	Step                  cstypes.RoundStepType
```

Comment:
lets be clear about the basic type here

Reply:
What do you mean? Should we define RoundStepType?It's somehow implementation detail and can change, so I am not sure if it should be part of spec.

Reply:
Let's just give a concrete type - int8

Reply:
You mean we shouldn't use int type? Because we use it in Vote so I thought that's fine.